rpc/server_interceptors.go

```diff
@@ -0,0 +1,126 @@
+package rpc
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/edaniels/golog"
+	"go.opencensus.io/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+// UnaryServerTracingInterceptor starts a new Span if Span metadata exists in the context.
+func UnaryServerTracingInterceptor(logger golog.Logger) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		remoteSpanContext, err := remoteSpanContextFromContext(ctx)
+		var span *trace.Span
+		if err == nil {
```

Comment:
[[style](https://viam.atlassian.net/wiki/spaces/ENG/pages/262371/Viam+Engineering+Guidelines)]: you probably want to do `err != nil` checks and fail-fast when it [reduces nesting](https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting) and limits the [scope of variables](https://github.com/uber-go/guide/blob/master/style.md#reduce-scope-of-variables) here and throughout. Line 30 looks like an obvious exception since it looks it would do the opposite.